### DISCUSSION TITLE
Lesson 7 : More Cheatcodes - failing tests depend on functions name ???

### DISCUSSION THREAD

**Discussion by Imad-Allal** [Cyfrin/foundry-full-course-cu#1250] [Q&A]
Lesson 7 : More Cheatcodes, of Patrick's course (On Cyfrin updraft: [Foundry Fundamentals] : 13. Foundry tests cheatcodes, 

Hey guys,
I did everything the author did in the first test, where the goal was to fail a test using the `vm.expectRevert()`, but I used another function name, and so my test didn't fail, I tried with the same name of function he used, and now it worked ! 
Does it have any relation with function naming ?
See the pictures below for more details
![Capture d'écran 2023-12-27 164427](https://github.com/Cyfrin/foundry-full-course-f23/assets/80546947/98e8b74a-7ebc-4bce-b9b0-91acd154cf47)
![Capture d'écran 2023-12-27 164543](https://github.com/Cyfrin/foundry-full-course-f23/assets/80546947/d5d729e3-20f0-42b9-a1d2-a8bd251b2d8d)

**Comment by ghost** [comment]
You can use the prefixes `test` or `testFail` to name your test functions.

TLDR:
- Prefixing with `test` indicates you're expecting the test to pass.
- Prefixing with `testFail` indicates you're expecting the test to fail.

 That's why when you changed the name to `testFundingFailsWithoutEnougthEth` the test it's expected to pass **due to the prefix**, but since you implemented the `expectRevert` cheatcode it's going to revert and the test will fail.

More details: https://book.getfoundry.sh/forge/cheatcodes#cheatcodes
